packages/core/content-type-builder/admin/src/components/FormModal/index.js

```diff
@@ -955,29 +956,51 @@ const FormModal = () => {
                 }}
               >
                 <Flex justifyContent="space-between">
-                  <Typography as="h2" variant="beta">
-                    {formatMessage(
-                      {
-                        id: getModalTitleSubHeader({
-                          actionType,
-                          forTarget,
-                          kind,
+                  {customField ? (
+                    <Typography as="h2" variant="beta">
+                      {formatMessage(
+                        {
+                          id: getModalTitleSubHeader({
+                            actionType,
+                            forTarget,
+                            kind,
+                            step,
+                            modalType,
+                          }),
+                          defaultMessage: 'Add new field',
+                        },
+                        {
+                          type: upperFirst(formatMessage(customField.intlLabel)),
+                          name: upperFirst(customField.name),
```

Comment:
This wasn't clear in the ticket but here we should pass `attributeName`, when editing an attribute using a custom field we should see "Edit {attributeName}" and when creating a new custom field we should "Add new {attributeType}" if that makes sense
```suggestion
                          name: upperFirst(attributeName),
```